### DISCUSSION TITLE
The wave cursor is not consistent, it doesnt move with other sound, but it does move with the others when playing the audio

### DISCUSSION THREAD

**Discussion by nel1991** [katspaugh/wavesurfer.js#3232] [Q&A]
![image](https://github.com/katspaugh/wavesurfer.js/assets/116612237/10f1a9a9-acd2-4084-8ad2-2ad3132ba77d)

On the photo above, you will see the 2 audios are working okay, the cursor is moving forward, but on the third one its not, it plays the sound, but the wave is not moving.

For the waveform, my code is: 

import React, { useState, useEffect, useRef } from 'react';
import PropTypes from 'prop-types';
import WaveSurfer from 'wavesurfer.js';
import styled from 'styled-components';
import { FaPlayCircle, FaPauseCircle } from 'react-icons/fa';
import { Howl } from "howler"
import Minimap from 'wavesurfer.js';

const Waveform = ({ audio }) => {
  const containerRef = useRef();
  const waveSurferRef = useRef({
    isPlaying: () => true,
  });
  const [isPlaying, toggleIsPlaying] = useState(true);
  
  useEffect(() => {
    const waveSurfer = WaveSurfer.create({
      container: containerRef.current,
      //container: '#waveform',
      responsive: true,
      barWidth:1,
      waveColor: 'white',
      progressColor: 'gray',
      barRadius: 4,
      responsive: true,
      height: 50,
      width: 30,
      barGap:0,
      pixelRatio: 5,
      barMinHeight:100,
      normalize: true,
      partialRender: true,
      //mediaControls: true,
      cursorWidth: 0,
      
     
    });
    waveSurfer.load(audio);
      waveSurfer.on('ready', () => {
      waveSurferRef.current = waveSurfer;
    });

    waveSurfer.on('finish', function () {
       toggleIsPlaying(true);
      // alert('done');
    });

  

    return () => {
      waveSurfer.destroy();
    };
  }, [audio]);

      

  return (
    <WaveSurferWrap>
      <button
        onClick={() => {
          waveSurferRef.current.playPause();
          toggleIsPlaying(waveSurferRef.current.isPlaying());
          toggleIsPlaying(!isPlaying);

        }}
        type="button"
      >
      
      { isPlaying ?  <FaPlayCircle size="3em" />: <FaPauseCircle size="3em" />}
       
      </button>
      <div ref={containerRef} />

    
      <div className="waveform__counter"> </div>
      <div className="waveform__duration"> </div>
   
    </WaveSurferWrap>
  );
};

Waveform.propTypes = {
  audio: PropTypes.string.isRequired,
};

const WaveSurferWrap = styled.div`
  display: grid;
  grid-template-columns: 40px 1fr;
  align-items: center;

  width: 550px;

  button {
    width: 40px;
    height: 40px;
    border: none;
    padding: 0;
    background-color:transparent;
    color:white;
    margin-right: 30px;
    cursor:pointer;
  }
`;

export default Waveform;

and for the display of the audio: 

<AllDatahere>	
					    <div className="taghere">					  

			    		{allDetails.map((allDetail, index) => {
					       return (
					          <div key={index} className="allsides">
					           	<div className="sideonedetail">
					           		<img src="../../miguel.jpg" width="70" height="50" className="imgphoto"/> 
					           		<h5 className="counthere"> <span className="spac"> </span> Song {index}
					           		</h5>
					           		
					            	<h5 className="tagdes">{allDetail.tags.join(' ')}</h5>
					           	</div>
					           	<div className="sidetwodetail">
					           		<Waveform audio={allDetail.previews['preview-hq-mp3']} />
					           	
					           		
					           	</div>
					          	<div className="downdiv">
					          		<button onClick={() => downloadResource(allDetail.previews['preview-hq-mp3'])} className="down"> <FiDownloadCloud size="2em" color="white" /> </button>
					          	</div>
					          </div>
					      	);
					      })}

			    		</div>
	    			}
	    	
	  
	    			</AllDatahere>

Any one can help me? Please

**Comment by katspaugh** [comment]
Try inserting a regular `<audio>` element instead of wavesurfer. Chances are it will also not move.

**Reply by nel1991** [reply]
Do you know what are the reasons why the cursor is not consisting in moving?

**Reply by katspaugh** [reply]
Could be the file format or your server configuration.
